test/Table_Tests/src/Common_Table_Operations/Filter_Spec.enso

```diff
@@ -435,21 +435,27 @@ spec setup =
             err2.should_fail_with Floating_Point_Equality
 
         Test.specify "should attach a warning when Nothing is used as a value in a comparison or `is_in` `Filter_Condition`" <|
-            t = table_builder [["x", [1, 2, 3, 4, 5]]]
+            t = table_builder [["x", [1, 2, 3]]]
             fcs = [Filter_Condition.Equal Nothing, Filter_Condition.Not_Equal Nothing]
                 + [Filter_Condition.Less Nothing, Filter_Condition.Equal_Or_Less Nothing]
                 + [Filter_Condition.Equal_Or_Greater Nothing, Filter_Condition.Greater Nothing]
                 + [Filter_Condition.Between Nothing Nothing , Filter_Condition.Is_In [Nothing]]
                 + [Filter_Condition.Not_In [Nothing]]
                 + [Filter_Condition.Is_In [1, Nothing, 2]]
             fcs.map fc->
-                Problems.expect_warning Nothing_Value_In_Filter_Condition (t.filter "x" fc . at "x")
+                Test.with_clue fc.to_text <|
+                    Problems.expect_warning Nothing_Value_In_Filter_Condition (t.filter "x" fc . at "x")
+
+        Test.specify "should not attach a warning when comparing with a column containing Nothing in a comparison or `is_in` `Filter_Condition`" <|
+            t = table_builder [["x", [1, 2, 3]], ["y", [1, Nothing, 2]]]
+            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Equal (t.at "y")))
 
         Test.specify "should not attach a warning when Nothing is not used as a value in a comparison or `is_in` `Filter_Condition`" <|
-            t = table_builder [["x", [1, 2, 3, 4, 5]]]
-            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Equal (t.at "x")))
-            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Equal 12) . at "x")
+            t = table_builder [["x", [1, 2, 3]], ["y", [1, Nothing, 2]]]
+            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Equal 12))
             Problems.assume_no_problems (t.filter "x" (Filter_Condition.Equal [Nothing, Nothing]))
+            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Is_In [[Nothing, Nothing]]))
+            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Is_In [(t.at "y")]))
```

Comment:
```suggestion
            Problems.assume_no_problems (t.filter "x" (Filter_Condition.Is_In (t.at "y")))
```
I think this should go this way. A column inside of a vector is 'too deep' nesting - it would check equality of each value from `x` if it is equal to _the whole column "y"_, that does not seem right.

Reply:
Done